src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -45,22 +48,35 @@ private void StartModelUpdateAndSelectedItemUpdateTasks(int modelUpdateDelay, in
 
             // Enqueue a new computation for the model
             var asyncToken = _asyncListener.BeginAsyncOperation(GetType().Name + ".StartModelUpdateTask");
-            _modelTask =
-                Task.Delay(modelUpdateDelay, cancellationToken)
-                    .SafeContinueWithFromAsync(
-                        _ => ComputeModelAsync(textSnapshot, cancellationToken),
-                        cancellationToken,
-                        TaskContinuationOptions.OnlyOnRanToCompletion,
-                        TaskScheduler.Default);
+            _modelTask = ComputeModelAfterDelayAsync(_modelTask, textSnapshot, modelUpdateDelay, cancellationToken);
```

Comment:
much simpler way to make a chain of tasks that allows async/await to be used.

Reply:
is it possible here for the task chain to get really long, so we end up queueing up a ton of these tasks for more and more outdated text snapshots?  E.g. moving the caret around faster than the delay so the task queue just keeps growing.  So by the time we make an edit it has to finish all the delayed tasks first since this waits for the previous task to complete before starting a new one.